### ISSUE TITLE
"Expose daemon on tcp://localhost:2375 without TLS" does not work with WSL mirrored mode networking

### ISSUE THREAD

**Issue by J0F3** [docker/for-win#13829]
### Description

When the "Expose daemon on tcp://localhost:2375 without TLS" option is enabled in the setting while WSL ist running with the mirrored mode networking it does not work as expected:
1. First, immediate after enabling the setting the port 2375 gets actually not exposed as expected. (Port is not reachable)
2. After quit and re-starting Docker Desktop is will not start again. Actually it starts and shuts down itself again without any error message. But the result is that Docker Desktop can not be started again. 

### Reproduce

1. Enable "mirror" networking in WSL2.
2. Start Docker and enable the "Expose daemon on tcp://localhost:2375 without TLS" option in the settings.
3. Quit Docker Desktop.
4. Start Docker Desktop again.

### Expected behavior

It should expose the docker daemon on tcp://localhost:2375 and should start correctly.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:08:44 2023
 OS/Arch:           windows/amd64
 Context:           default

And in WSL:
Client: Docker Engine - Community
 Cloud integration: v1.0.35+desktop.5
 Version:           24.0.7
 API version:       1.43
 Go version:        go1.20.10
 Git commit:        afdd53b
 Built:             Thu Oct 26 09:08:02 2023
 OS/Arch:           linux/amd64
 Context:           default
```

### docker info

```bash
Client:
 Version:    24.0.7
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.0-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.23.3-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  0.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.10
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.2.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 4
 Server Version: 24.0.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d8f198a4ed8892c764191ef7b3b06d8a2eeb5c7f
 runc version: v1.1.10-0-g18a0cb0
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 15.62GiB
 Name: docker-desktop
 ID: 3c739c04-ef69-4837-b303-222ceacd324f
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

90FFFBA9-26C5-49C7-929C-7A1C5AA6F553/20231206103022

### Additional Info

Under %LOCALAPPDATA%\Docker there is file backend.error.json withe the following content:

```JSON
{
    "type": "*errordef.dockerDesktopError",
    "component": "backend",
    "originalError": {
        "type": "*fmt.wrapError",
        "error": {
            "type": "*errordef.dockerDesktopError",
            "component": "engines",
            "originalError": {
                "type": "*fmt.wrapError",
                "error": {
                    "type": "*errordef.dockerDesktopError",
                    "component": "engines",
                    "originalError": {
                        "type": "*fmt.wrapError",
                        "error": {
                            "type": "*errordef.dockerDesktopError",
                            "component": "vm:lifecycleserver",
                            "genericError": {
                                "type": "*errors.errorString",
                                "description": "fatal"
                            },
                            "originalError": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*fmt.wrapError",
                                    "error": {
                                        "type": "*fmt.wrapError",
                                        "error": {
                                            "type": "*errors.errorString",
                                            "description": "address already in use"
                                        },
                                        "message": "bind"
                                    },
                                    "message": "listen tcp 0.0.0.0:2375"
                                },
                                "message": "create dockerd TCP listener"
                            },
                            "creationTime": 1701854733035407872,
                            "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                            "creationLine": 73
                        },
                        "message": "failed to start VM - received state failed to start"
                    },
                    "creationTime": 1701854733672718600,
                    "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
                    "creationLine": 417,
                    "wrappedError": {
                        "type": "*errordef.dockerDesktopError",
                        "component": "vm:lifecycleserver",
                        "genericError": {
                            "type": "*errors.errorString",
                            "description": "fatal"
                        },
                        "originalError": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*fmt.wrapError",
                                    "error": {
                                        "type": "*errors.errorString",
                                        "description": "address already in use"
                                    },
                                    "message": "bind"
                                },
                                "message": "listen tcp 0.0.0.0:2375"
                            },
                            "message": "create dockerd TCP listener"
                        },
                        "creationTime": 1701854733035407872,
                        "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                        "creationLine": 73
                    }
                },
                "message": "starting vm"
            },
            "creationTime": 1701854733672718600,
            "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
            "creationLine": 178,
            "wrappedError": {
                "type": "*errordef.dockerDesktopError",
                "component": "engines",
                "originalError": {
                    "type": "*fmt.wrapError",
                    "error": {
                        "type": "*errordef.dockerDesktopError",
                        "component": "vm:lifecycleserver",
                        "genericError": {
                            "type": "*errors.errorString",
                            "description": "fatal"
                        },
                        "originalError": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*fmt.wrapError",
                                    "error": {
                                        "type": "*errors.errorString",
                                        "description": "address already in use"
                                    },
                                    "message": "bind"
                                },
                                "message": "listen tcp 0.0.0.0:2375"
                            },
                            "message": "create dockerd TCP listener"
                        },
                        "creationTime": 1701854733035407872,
                        "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                        "creationLine": 73
                    },
                    "message": "failed to start VM - received state failed to start"
                },
                "creationTime": 1701854733672718600,
                "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
                "creationLine": 417,
                "wrappedError": {
                    "type": "*errordef.dockerDesktopError",
                    "component": "vm:lifecycleserver",
                    "genericError": {
                        "type": "*errors.errorString",
                        "description": "fatal"
                    },
                    "originalError": {
                        "type": "*fmt.wrapError",
                        "error": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*errors.errorString",
                                    "description": "address already in use"
                                },
                                "message": "bind"
                            },
                            "message": "listen tcp 0.0.0.0:2375"
                        },
                        "message": "create dockerd TCP listener"
                    },
                    "creationTime": 1701854733035407872,
                    "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                    "creationLine": 73
                }
            }
        },
        "message": "starting engines"
    },
    "creationTime": 1701854733672718600,
    "creationFile": "common/cmd/com.docker.backend/main.go",
    "creationLine": 471,
    "wrappedError": {
        "type": "*errordef.dockerDesktopError",
        "component": "engines",
        "originalError": {
            "type": "*fmt.wrapError",
            "error": {
                "type": "*errordef.dockerDesktopError",
                "component": "engines",
                "originalError": {
                    "type": "*fmt.wrapError",
                    "error": {
                        "type": "*errordef.dockerDesktopError",
                        "component": "vm:lifecycleserver",
                        "genericError": {
                            "type": "*errors.errorString",
                            "description": "fatal"
                        },
                        "originalError": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*fmt.wrapError",
                                    "error": {
                                        "type": "*errors.errorString",
                                        "description": "address already in use"
                                    },
                                    "message": "bind"
                                },
                                "message": "listen tcp 0.0.0.0:2375"
                            },
                            "message": "create dockerd TCP listener"
                        },
                        "creationTime": 1701854733035407872,
                        "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                        "creationLine": 73
                    },
                    "message": "failed to start VM - received state failed to start"
                },
                "creationTime": 1701854733672718600,
                "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
                "creationLine": 417,
                "wrappedError": {
                    "type": "*errordef.dockerDesktopError",
                    "component": "vm:lifecycleserver",
                    "genericError": {
                        "type": "*errors.errorString",
                        "description": "fatal"
                    },
                    "originalError": {
                        "type": "*fmt.wrapError",
                        "error": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*errors.errorString",
                                    "description": "address already in use"
                                },
                                "message": "bind"
                            },
                            "message": "listen tcp 0.0.0.0:2375"
                        },
                        "message": "create dockerd TCP listener"
                    },
                    "creationTime": 1701854733035407872,
                    "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                    "creationLine": 73
                }
            },
            "message": "starting vm"
        },
        "creationTime": 1701854733672718600,
        "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
        "creationLine": 178,
        "wrappedError": {
            "type": "*errordef.dockerDesktopError",
            "component": "engines",
            "originalError": {
                "type": "*fmt.wrapError",
                "error": {
                    "type": "*errordef.dockerDesktopError",
                    "component": "vm:lifecycleserver",
                    "genericError": {
                        "type": "*errors.errorString",
                        "description": "fatal"
                    },
                    "originalError": {
                        "type": "*fmt.wrapError",
                        "error": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*fmt.wrapError",
                                "error": {
                                    "type": "*errors.errorString",
                                    "description": "address already in use"
                                },
                                "message": "bind"
                            },
                            "message": "listen tcp 0.0.0.0:2375"
                        },
                        "message": "create dockerd TCP listener"
                    },
                    "creationTime": 1701854733035407872,
                    "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                    "creationLine": 73
                },
                "message": "failed to start VM - received state failed to start"
            },
            "creationTime": 1701854733672718600,
            "creationFile": "common/cmd/com.docker.backend/internal/engines/engines.go",
            "creationLine": 417,
            "wrappedError": {
                "type": "*errordef.dockerDesktopError",
                "component": "vm:lifecycleserver",
                "genericError": {
                    "type": "*errors.errorString",
                    "description": "fatal"
                },
                "originalError": {
                    "type": "*fmt.wrapError",
                    "error": {
                        "type": "*fmt.wrapError",
                        "error": {
                            "type": "*fmt.wrapError",
                            "error": {
                                "type": "*errors.errorString",
                                "description": "address already in use"
                            },
                            "message": "bind"
                        },
                        "message": "listen tcp 0.0.0.0:2375"
                    },
                    "message": "create dockerd TCP listener"
                },
                "creationTime": 1701854733035407872,
                "creationFile": "linuxkit/pkg/init/cmd/lifecycle-server/main.go",
                "creationLine": 73
            }
        }
    }
}
```

**Comment by J0F3** [comment]
This seems to be fixed with the latest WSL Version ([2.0.14)](https://github.com/microsoft/WSL/releases/tag/2.0.14).

From the change log:
"Allow port binds from different network namespaces when mirrored networking is enabled."

Seems that this is probably the change which fixes this problem.

**Comment by MihaelaStoica** [comment]
Closed, as the issue is fixed in the latest WSL version.
